code/game/objects/items/toys/toys.dm

```diff
@@ -538,6 +542,217 @@
 	. = ..()
 	AddComponent(/datum/component/squeak, 'sound/items/gnome.ogg', 50)
 
+/obj/item/toy/plush/gnome/living
+	resistance_flags = UNACIDABLE
+	///how far the gnome should choose for teleportation purposes
+	var/gnome_move_range = 5
+	///how many failed teleports we've done
+	var/teleport_retries = 0
+	///sanity cap to prevent gnome spending too much time calculating possible teleport areas, it's theoretically possible to store the gnome in an impossible area so we need to check this
+	var/max_tries = 150
+	///list for keeping track of the mobs around us
+	var/mob/possible_mobs = list()
+	///list for keeping track of items in current gnome turf
+	var/list/turf_contents = list()
+	///used for determining if a gnome is in the pipe network
+	var/pipe_mode = FALSE
+	///how likely are we to enter a pipe
+	var/pipe_mode_chance = STANDARD_GNOME_PIPE_CHANCE
+	///hold an int that determines what the interval a gnome acts
+	var/gnome_act_timer
+	///original gnome spawn location, used as an emergency backup
+	var/gnome_origin
+
+/obj/item/toy/plush/gnome/living/Initialize(mapload)
+	. = ..()
+	gnome_origin = get_turf(src)
+	addtimer(CALLBACK(src, PROC_REF(gnome_act)), 5 MINUTES)
+	RegisterSignal(src, COMSIG_MOVABLE_SHUTTLE_CRUSH, PROC_REF(shuttle_crush))
+
+///root proc for handling gnome AI routines
+/obj/item/toy/plush/gnome/living/proc/gnome_act()
+	turf_contents = list()
+	///targetturf stores our position at the start of routine in all. teleport_routine uses it as a baseline for determining where to find teleport locations also
+	var/turf/targetturf = get_turf(src)
+	for(var/atom/movable/object AS in targetturf.contents)
+		turf_contents += object
+	gnome_act_timer = rand(4,8)
+	pipe_mode_chance = STANDARD_GNOME_PIPE_CHANCE
+	if(prob(10))
+		do_flavor_actions(targetturf) //flavor actions don't take being watched into account
+		addtimer(CALLBACK(src, PROC_REF(gnome_act)), gnome_act_timer MINUTES)
+		return
+	possible_mobs = list()
+	for(var/mob/nearbymob in range(GNOME_EXCLUSION_RANGE, src)) //all mobs are included except animals, this means that AIs and ghosts will block gnome movement
+		if(isanimal(nearbymob))
+			continue
+		possible_mobs += nearbymob
+	if(handle_pipe_mode(targetturf) && pipe_mode) //if we're in pipe_mode we return,
+		addtimer(CALLBACK(src, PROC_REF(gnome_act)), gnome_act_timer MINUTES)
```

Comment:
just make gnome_act_timer in ticks

Reply:
I don't actually know what you mean by this, do you mean like gnome_act_timer = 4000 instead of something like gnome_act_timer = 5 MINUTES?